src/wgpu/wgpu-surface.cpp

```
@@ -167,39 +167,40 @@ Result SurfaceImpl::init(DeviceImpl* device, WindowHandle windowHandle)
-        return SLANG_FAIL;
+        resolvedConfig.format = m_info.preferredFormat;
     }
-    if (m_config.format == Format::Undefined)
+    if (resolvedConfig.usage == TextureUsage::None)
     {
-        m_config.format = m_info.preferredFormat;
-    }
-    if (m_config.usage == TextureUsage::None)
-    {
-        m_config.usage = m_info.supportedUsage;
+        resolvedConfig.usage = (TextureUsage::Present | TextureUsage::RenderTarget | TextureUsage::CopyDestination) &
```

Cppcheck (2.21.0)

[style] The function 'rbegin' is never used.

(unusedFunction)